### PR TITLE
Nac 1902 chk email

### PR DESCRIPTION
**Pull request by kcherukupalli**
## What’s changing?
Validate the status of the State for sending Notifications.
 DuplicateParticipant:   
-     No notifications if the Initiating state is disabled.
-     No notification if the Matching state is disabled, send notification to Initiating state if Initiating state is enabled.
Match Resolution:
  - No notifications if disabled state is involved.

IAC:
  Updated MatchRes Function and State info function with "enabledStates" configuration.  

## Why?

NAC-1902

## This PR has:

- [x] **JIRA ticket(s)** (_Must be listed in PR request and branch name should follow convention NAC-123-description_)<description>

- [ ] **Security Related Changes** (_Security Tab questions have been answered on JIRA ticket(s)_)

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Documented API changes** (_required on public classes and methods, and elsewhere as appropriate_)
    - [ ] External API documentation - OpenApi docs and Generated Docs (e.g. https://github.com/18F/piipan/tree/dev/docs/openapi) 
    - [ ] Internal API documentation (e.g. https://github.com/18F/piipan/tree/dev/metrics/docs/openapi)   

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)
    - [x] Have IaC scripts run successfully against an existing deployed environment?
    - [x] Have IaC scripts run successfully against a newly deployed environment?
    - [ ] Have any new bash scripts been given the appropriate permissions? (e.g. git update-index --chmod=+x $script)
    - [ ] Do the IaC changes require deleting existing resources?

## Anything else?

_Add other details that are helpful for review here_

## Code Review Checklist:

- [x] Automated checks all pass? (code builds successfully, tests pass, test coverage meets/exceeds threshold, etc.) 
- [x] Have all requirements and Acceptance Criteria been met?
- **If there are code changes**
    - [x] Is there no unnecessary complexity? Code is readable, maintainable, etc.
    - [ ] Do the changes follow SOLID principles?
    - [ ] Are the changes robust? Concurrency considered? Performance? Security (e.g. no sql injections)?
    - [ ] Are newly added dependencies needed and appropriate? Are dependency files (e.g. packages.json & packages.lock.json) updated?
    - [ ] For any new projects/solutions, have they been added to code coverage files?
    - [ ] Public classes & methods are documented    
- **If there are API changes**
    - [ ] Does API make sense? Useful and concise?
    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
    - [ ] Are API tests included?
    - [ ] Are API responses appropriate and documented? (200s, 201, 404s, etc.)
    - [ ] Are internals kept private and not leaked into the API?
    - [ ] External API documentation - OpenApi docs and Generated Docs 
    - [ ] Internal API documentation    
- **If there are IAC changes**
    - [ ] IAC Scripts are readable & maintainable
    - [ ] IAC Scripts are documented
    - [ ] IAC scripts pass required checks
    - [ ] IAC scripts were run successfully by author
- **General Logging Questions**
    - [ ] Do new capabilities (or changed capabilities) include logging?
    - [ ] Is PII not logged?
- **General Error Handling Questions**
    - [ ] Has exception handling been added for new capabilities (or changed capabilities)?
    - [ ] Are errors/exceptions properly communicated to end-users? Were there requirements for user facing error messages?
    - [ ] Are errors/exceptions logged?
    - [ ] Is PII kept out of error/exception logging?
- **If there are Developer Documentation Changes**
    - [ ] General Project documentation (under piipan\docs. e.g. iac.md, architecture.md, etc.)
    - [ ] Github documentation additions/changes (Readmes)
- **General automated test questions**
    - [ ] End-to-End Tests provided (if code implements new user story)
    - [ ] Are tests sufficient?
    - [ ] Corner cases considered?
    - [ ] Are performance tests needed?

### CODE REVIEW

File: match/src/Piipan.Match/Piipan.Match.Core/Services/MatchEventService.cs [hunk 1]
```diff
@@ -160,22 +162,27 @@ private async Task<ParticipantMatch> CreateNewOrReturnExistingMatch(IParticipant
...
+                notificationRecord.IsInitiatingStateEnabled = enabledStatesList.Contains(initState.StateAbbreviation.ToLower());
 
                 notificationRecord.MatchingStateEmailRecipientsModel = new EmailToModel();
                 notificationRecord.MatchingStateEmailRecipientsModel.EmailTo = matchingState?.Email;
                 notificationRecord.MatchingStateEmailRecipientsModel.EmailCcTo = matchingState?.EmailCc;
+                notificationRecord.IsMatchingStateEnabled = enabledStatesList.Contains(matchingState.StateAbbreviation.ToLower());
                 notificationRecord.MatchEmailDetails.ReplyToEmail = Environment.GetEnvironmentVariable("SmtpReplyToEmail");
-
-                await _notificationService.PublishNotificationOnMatchCreation(notificationRecord); //Publishing Email for Initiating & Matching State:  Based on the requirement
+                if (notificationRecord.IsInitiatingStateEnabled) //No Email is sent out if the Initiating State is not Enabled // Send emails only if the state is enabled. nac-1902
```
Comment (bgfoshay) [inline match/src/Piipan.Match/Piipan.Match.Core/Services/MatchEventService.cs:184]: Question: What will happen here if IsInitiatingStateEnabled==true while IsMatchingStateEnabled==false?

File: notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs [hunk 2]
```diff
@@ -32,8 +37,15 @@ public async Task<bool> PublishNotificationOnMatchCreation(NotificationRecord no
             var emailbodyMS = await _viewRenderService.GenerateMessageContent("MatchEmail.cshtml", notificationRecord.MatchEmailDetails);
             var emailsubjectMS = string.Format(subject, "Matching State", notificationRecord.MatchEmailDetails.InitState);
 
-            var resultIS = await PublishNotifications(notificationRecord.InitiatingStateEmailRecipientsModel, emailbodyIS, emailsubjectIS);
-            var resultMS = await PublishNotifications(notificationRecord.MatchingStateEmailRecipientsModel, emailbodyMS, emailsubjectMS);
+            var resultIS = false;
+            var resultMS = false;
+
+            resultIS = await PublishNotifications(notificationRecord.InitiatingStateEmailRecipientsModel, emailbodyIS, emailsubjectIS);
```
Comment (bgfoshay) [inline notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs:43]: Question: Does there need to be a check before publishing for notificationRecord.IsInitiatingStateEnabled?

Comment (bgfoshay) [reply notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs:43]: Nm. I see you check that up above.

File: notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs [hunk 3]
```diff
@@ -20,8 +20,13 @@ public NotificationService(INotificationPublish notificationPublish, IViewRender
 
         public async Task<bool> PublishNotificationOnMatchCreation(NotificationRecord notificationRecord)
         {
+            
             string subject = $"{{0}}: NAC Match {notificationRecord.MatchEmailDetails.MatchId} with {{1}}";
 
+            // Send emails only if the state is enabled. nac-1902
+            if (!notificationRecord.IsInitiatingStateEnabled) //No Email is sent out if the Initiating State is not Enabled
```
Comment (bgfoshay) [inline notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs:27]: Question: should this function exit early if notificationRecord.IsMatchingStateEnabled if false?

File: match/src/Piipan.Match/Piipan.Match.Core/Services/MatchEventService.cs (hunk 1 above)
Comment (kcherukupalli) [reply match/src/Piipan.Match/Piipan.Match.Core/Services/MatchEventService.cs:184]: No notification is sent to MatchingState.  It is checking in the 'PublishNotificationOnMatchCreation'.  Whereas a notification will be sent to Initiating state.

File: notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs (hunk 3 above)
Comment (kcherukupalli) [reply notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs:27]: Need to send notification for Initiating State which is enabled.

**Review by bgfoshay — APPROVED** [review]
LGTM! Thanks for updating the logic!

**Pull request merged**